### PR TITLE
CI macOS build fixes.

### PR DESCRIPTION
**Pull request by vrcamillo**
Solved macOS CI fails. 

Set python version to be lower than 3.11 while some of the dependencies (toml) update support and also to solve macOS build problems.

Fixes cvc5/cvc5-projects/issues/534

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -494,7 +494,7 @@ if(BUILD_BINDINGS_PYTHON AND NOT BUILD_SHARED_LIBS)
   set(BUILD_BINDINGS_PYTHON OFF)
 endif()
 if(BUILD_BINDINGS_PYTHON)
-  set(BUILD_BINDINGS_PYTHON_VERSION ${PYTHON_VERSION_MAJOR})
```
Comment (mpreiner) [inline CMakeLists.txt:497]: Is this change necessary? `${PYTHON_VERSION_MAJOR}` should be 3.

File: src/api/python/CMakeLists.txt [hunk 2]
```diff
@@ -53,7 +53,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND
   # versions of CMake. If we are compiling cvc5 on macOS and we have a newer
   # version of CMake available, we try to use find_package(Python ...) to find
   # the paths.
-  find_package(Python COMPONENTS Development)
+  find_package(Python 3.10 EXACT COMPONENTS Development)
```
Comment (mpreiner) [inline src/api/python/CMakeLists.txt:56]: cvc5 supports Python 3.6 up to 3.10, so we should not force our users to use an exact version here. We could do a range check here `3.6...<3.11` instead of 3.10 and add a comment why we exclude 3.11 for now.

Comment (vrcamillo) [reply src/api/python/CMakeLists.txt]: Fixed.

File: CMakeLists.txt (hunk 1 above)
Comment (vrcamillo) [reply CMakeLists.txt]: Fixed.

**Comment by vrcamillo** [comment]
@mpreiner I changed the details you mentioned.

This will solve issue cvc5-projects#534

File: src/api/python/CMakeLists.txt [hunk 3]
```diff
@@ -53,7 +53,10 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND
   # versions of CMake. If we are compiling cvc5 on macOS and we have a newer
   # version of CMake available, we try to use find_package(Python ...) to find
   # the paths.
-  find_package(Python COMPONENTS Development)
+
+  # As a temporary workaround, we do not support python 3.11 due to the problem described above.
+  # Once this issue (https://github.com/actions/setup-python/issues/577) is solved, we will be able to add support to version 3.11.
```
Comment (mpreiner) [inline src/api/python/CMakeLists.txt:58]:
```suggestion
  # Once this issue (https://github.com/actions/setup-python/issues/577) is
  # solved, we will be able to add support to version 3.11.
```
80 char lines

**Review by mpreiner — APPROVED** [review]
LGTM, thanks for the changes!
